parallel/runner.go

```diff
@@ -237,17 +185,22 @@ func (r *runner) addThread() {
 	nextThreadId := atomic.AddUint32(&r.threadCount, 1) - 1
 	go func(threadId int) {
 		defer r.threadsWaitGroup.Done()
-
-		r.runningThreadsLock.Lock()
-		r.runningThreads++
-		r.runningThreadsLock.Unlock()
+		r.openThreadsLock.Lock()
+		r.openThreads++
+		r.openThreadsLock.Unlock()
 
 		// Keep on taking tasks from the queue.
 		for t := range r.tasks {
-			r.idle.Store(threadId, false)
-
+			// Increase num of threads running tasks
+			atomic.AddUint32(&r.activeThreads, 1)
+			// Mark that some task was executed and the runner started with tasks executing.
```

Comment:
Not sure the phrasing of thsi comment is the best. Anyway, I think a description for the below code is not needed.

Reply:
done